_posts/2019-04-05-omakase-typescript.markdown

```diff
@@ -0,0 +1,179 @@
+---
+layout: epic
+title: "What is TypeScript?"
+date: "2019-05-05"
+author: [orta]
+categories: [community, javascript, roads and bridges, typescript]
+series: Omakase
+---
+
+TypeScript is a language from Microsoft which builds on JavaScript. This post is a non-technical overview of what
+JavaScript is, how TypeScript extends it and why we choose to adopt it at Artsy.
+
+<!-- more -->
+
+## What is JavaScript?
+
+First up, you can't describe TypeScript without talking about JavaScript. To create a website (and a bunch of other
+types of things) you work in three languages: HTML, CSS and JavaScript (JS). Broadly speaking: HTML defines the
+content that will appear on the page, CSS defines the visual style of the page, and JS defines the interactive
+behaviours of the page.
+
+We describe having these sets of skills as being a "front-end" developer. You have to understand those three
+languages to present anything inside a web browser like Safari, Firefox or Chrome. So, given how popular the web
+is, there is a massive demand for people who are good at using these three languages.
+
+There is also the set of skills for the "back-end" developers, which are to create computer services that
+communicate either to a web browser (by passing it HTML/CSS/JS) or to another service (by sending a simpler form of
+JavaScript.) You don't need to use HTML, CSS or JS to write this type of code, but it's usually an end-product of
+your work. We mostly build our back-ends in Ruby or JavaScript at Artsy.
+
+### What do Programming Languages do?
+
+Programming languages are an interesting problem to solve. People read code many, many multiples of times more than
+they write it - so developers create languages which are good at solving particular problems with a small amount of
+code. Here's an example using JavaScript:
+
+```js
+var name = "Danger"
+console.log("Hello, " + danger)
+```
+
+The first line makes a variable (a kind of box you can keep things in) and then the second line outputs text to the
+console (think DOS, or the terminal) `"Hello, Danger"`. JavaScript is designed to work as a scripting language,
+which means the code starts at the top of the file and then goes through line by line. To provide some contrast,
+here is the [same behavior](https://repl.it/repls/VioletredGlisteningInfo) in Java, which is built with different
+language constraints:
+
+```java
+class Main {
+  public static void main(String[] args) {
+    String name = "Danger";
+    System.out.println("Hello, " + name);
+  }
+}
+```
+
+> Note: if you find the naming of Java and JavaScript confusing (it is, they are two completely separate
+> programming languages, no link at all.) JavaScript was named that way because Java was looking to be really the
+> next hot language (it did turn out that way for a decade or two, but now JavaScript is usually the first language
+> people have heard of.)
```

Comment:
This could link out to a resources to learn more, maybe this one? https://medium.com/@benastontweet/lesson-1a-the-history-of-javascript-8c1ce3bffb17